tools/aa/commands/WorkspaceCommand.java

```diff
@@ -42,93 +43,162 @@
+    if (guessedRepoFolder != null) {
+      System.out.println(String.format("cd %s", guessedRepoFolder));
+    } else {
+      System.out.println(String.format("cd %s", workspacePath));
+    }
+    return true;
+  }
 
+  private boolean processArgs(String[] args) {
     // The Flags library does not support short flags by design, as this increases the chance of
     // flag collisions between packages. To allow short flags here, we do a flag replacement.
     for (int i = 0; i < args.length; i++) {
       if (args[i].equals("-f")) {
         args[i] = "--force";
+      } else if (args[i].equals("-r")) {
+        args[i] = "--remove";
       }
     }
     if (args.length < 2) {
       System.err.println(RED_ERROR + "Missing workspace name");
       return false;
     }
 
-    String workspaceName = args[args.length - 1];
+    workspaceName = args[args.length - 1];
+    workspacePath = fileUtils.joinPaths(config.getBasePath(), "ws", workspaceName);
+
     if (workspaceName.startsWith("-")) {
       System.err.println(RED_ERROR + "Missing workspace name");
       return false;
     }
 
     args = Arrays.copyOfRange(args, 0, args.length - 1);
-
     Flags.parse(args, WorkspaceCommand.class.getPackage());
-    String basePath = config.getBasePath();
-    String workspacePath = fileUtils.joinPaths(basePath, "ws", workspaceName);
+
+    if (force.get() && remove.get()) {
+      System.err.println(RED_ERROR + "Conflicting options specified");
+      return false;
+    }
 
     if (force.get()) {
-      if (fileUtils.folderExists(workspacePath)) {
-        System.err.println(RED_ERROR + "Workspace already exists");
+      mode = ActionMode.CREATE;
+    } else if (remove.get()) {
+      mode = ActionMode.REMOVE;
+    } else {
+      mode = ActionMode.SWITCH;
+    }
+
+    return true;
+  }
+
+  // Guess folder to cd into after workspace command
+  private String guessFolderForCd(String workspacePath) {
+    // Try using current workspace subfolder
+    String guessedRepoFolder =
+        fileUtils.joinPaths(workspacePath, fileUtils.getCurrentWorkingDirectoryName());
+    if (fileUtils.folderExists(guessedRepoFolder)) {
+      return guessedRepoFolder;
+    }
+    // If target workspace has only 1 subfolder, use it
+    try {
+      if (fileUtils.listContents(workspacePath).size() == 1) {
+        guessedRepoFolder =
+            fileUtils.joinPaths(workspacePath, fileUtils.listContents(workspacePath).get(0));
+        if (fileUtils.folderExists(guessedRepoFolder)) {
+          return guessedRepoFolder;
+        }
+      }
+    } catch (Exception e) {
+      // Fail silently. It's just a nice-to-have.
+    }
+    return null;
+  }
+
+  @Override
+  public boolean run(String[] args) {
+    // Note: System.out gets executed by the calling aa_tool.sh to run commands such as cd.
+    if (!processArgs(args)) {
+      return false;
+    }
+
+    if (mode == ActionMode.CREATE) {
+      if (!createWorkspace()) {
```

Comment:
How about we put the positive first:
```
if (createWorkspace()) {
  switchWorkspace();
} else {
  return false;
}
```
(also below)

Reply:
yay, we saved space on two `!` characters 🙂

Reply:
:-)